Pachctl transactions returns runtime error

<!-- This form is for bug reports ONLY! 

If you're looking for help, please check:

Docs: http://docs.pachyderm.io/en/latest/
Slack: http://slack.pachyderm.io/

-->

**What happened?**:
I was trying to delete a transaction via `pachctl delete transaction <transactionid>` and the command returned the following:

```
pachctl delete transaction ea7e71b4-7933-4050-b63d-100efe42b053
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x23cdb7e]

goroutine 1 [running]:
github.com/pachyderm/pachyderm/src/server/transaction/cmds.Cmds.func5(0xc00018c870, 0x1, 0x1, 0x0, 0x0)
        src/server/transaction/cmds/cmds.go:193 +0x12e
github.com/pachyderm/pachyderm/src/server/pkg/cmdutil.RunBoundedArgs.func1(0xc0004db680, 0xc00018c870, 0x1, 0x1)
        src/server/pkg/cmdutil/cobra.go:38 +0x173
github.com/pachyderm/pachyderm/src/server/vendor/github.com/spf13/cobra.(*Command).execute(0xc0004db680, 0xc00018c830, 0x1, 0x1, 0xc0004db680, 0xc00018c830)
        src/server/vendor/github.com/spf13/cobra/command.go:572 +0x24a
github.com/pachyderm/pachyderm/src/server/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00032e480, 0x0, 0x0, 0x0)
        src/server/vendor/github.com/spf13/cobra/command.go:658 +0x2cc
github.com/pachyderm/pachyderm/src/server/vendor/github.com/spf13/cobra.(*Command).Execute(0xc00032e480, 0x299e6e0, 0x299e6e8)
        src/server/vendor/github.com/spf13/cobra/command.go:617 +0x2b
main.main.func1(0x0, 0x0)
        src/server/cmd/pachctl/main.go:20 +0x4b
main.main()
        src/server/cmd/pachctl/main.go:21 +0xc8
```

**What you expected to happen?**:
The command should have succeded 

**How to reproduce it (as minimally and precisely as possible)?**:
Create a transaction then try to delete it

**Anything else we need to know?**:

**Environment?**:
- Kubernetes version (use `kubectl version`): 1.15.0
- Pachyderm CLI version (use `pachctl version`): 1.9.0
- Cloud provider (e.g. aws, azure, gke) or local deployment (e.g. minikube vs dockerized k8s): local
- OS (e.g. from /etc/os-release): 10.13.6
- Others:

--- a/src/server/transaction/cmds/cmds.go
+++ b/src/server/transaction/cmds/cmds.go
@@ -182,14 +182,16 @@ transaction' or cancelled with 'delete transaction'.`,
 			// TODO: use advisory locks on config so we don't have a race condition if
 			// two commands are run simultaneously
 			var txn *transaction.Transaction
-			var isActive bool
+			isActive := false
 			if len(args) > 0 {
 				txn = &transaction.Transaction{ID: args[0]}
 
 				// Don't check err here, this is just a quality-of-life check to clean
 				// up the config after a successful delete
 				activeTxn, _ := requireActiveTransaction()
-				isActive = (txn.ID == activeTxn.ID)
+				if activeTxn != nil {
+					isActive = txn.ID == activeTxn.ID
+				}
 			} else {
 				txn, err = requireActiveTransaction()
 				if err != nil {

--- a/src/server/transaction/cmds/cmds_test.go
+++ b/src/server/transaction/cmds/cmds_test.go
@@ -1,6 +1,7 @@
 package cmds
 
 import (
+	"fmt"
 	"os/exec"
 	"strings"
 	"testing"
@@ -42,4 +43,54 @@ func TestTransaction(t *testing.T) {
 		"repo", repo,
 		"commit", commit,
 	).Run())
+}
+
+func startTransaction(t *testing.T) string {
+	output, err := (*exec.Cmd)(tu.BashCmd("pachctl start transaction")).Output()
+	require.NoError(t, err)
+
+	strs := strings.Split(strings.TrimSpace(string(output)), " ")
+	return strs[len(strs)-1]
+}
+
+func requireTransactionDoesNotExist(t *testing.T, txn string) {
+	output, err := (*exec.Cmd)(tu.BashCmd("pachctl inspect transaction {{.txn}} 2>&1", "txn", txn)).Output()
+	require.YesError(t, err)
+	expected := fmt.Sprintf("%s not found", txn)
+	require.True(t, strings.Contains(string(output), expected))
+}
+
+func TestDeleteActiveTransaction(t *testing.T) {
+	// Start then delete a transaction
+	txn := startTransaction(t)
+	require.NoError(t, tu.BashCmd("pachctl delete transaction").Run())
+
+	// Check that the transaction no longer exists
+	requireTransactionDoesNotExist(t, txn)
+}
+
+func TestDeleteInactiveTransaction(t *testing.T) {
+	// Start, stop, then delete a transaction
+	txn := startTransaction(t)
+	require.NoError(t, tu.BashCmd("pachctl stop transaction").Run())
+	require.NoError(t, tu.BashCmd("pachctl delete transaction {{.txn}}", "txn", txn).Run())
+
+	// Check that the transaction no longer exists
+	requireTransactionDoesNotExist(t, txn)
+}
+
+func TestDeleteSpecificTransaction(t *testing.T) {
+	// Start two transactions, delete the first one, make sure the correct one is deleted
+	txn1 := startTransaction(t)
+	txn2 := startTransaction(t)
+
+	// Check that both transactions exist
+	require.NoError(t, tu.BashCmd("pachctl inspect transaction {{.txn}}", "txn", txn1).Run())
+	require.NoError(t, tu.BashCmd("pachctl inspect transaction {{.txn}}", "txn", txn2).Run())
+
+	require.NoError(t, tu.BashCmd("pachctl delete transaction {{.txn}}", "txn", txn1).Run())
+
+	// Check that only the second transaction exists
+	requireTransactionDoesNotExist(t, txn1)
+	require.NoError(t, tu.BashCmd("pachctl inspect transaction {{.txn}}", "txn", txn2).Run())
 }

